### ISSUE TITLE
Python4Delphi: issue when adding many methods to a module, might be related to ReallocMethods

### ISSUE THREAD

**Issue by jokli** [pyscripter/python4delphi#389]
I have a Python4Delphi project that registers 9 Delphi methods:

   ```
AddDelphiMethod('IsParcelBlocked',Scanplus_IsBlocked,'Checks if there is a blocking rule in SphV2');
AddDelphiMethod('DisplayMessage',Scanplus_Message,'Shows a Message in Scan+');
AddDelphiMethod('HasParcelHandlingRule',Scanplus_HasParcelHandlingRule,'Checks if there is a rule of a certain type in SphV2');
AddDelphiMethod('GetCurrentWeight',Scanplus_GetCurrentWeight,'Retrieve the weight of the current parcel');
AddDelphiMethod('GetExceptionCodes',Scanplus_GetExceptioncodes,'Retrieve a list of the current exceptioncodes');
AddDelphiMethod('GetUserInfo',Scanplus_GetUserInfo,'Retrieve information about the current user');
AddDelphiMethod('ReportSphRuleAsHandled',Scanplus_ReportRuleAsHandledByGUID,'Notify SPHV2 that the rule has been handled');
AddDelphiMethod('GetCurrentForm',Scanplus_GetCurrentForm,'Retrieve an object representing the current form with all user entries');
AddDelphiMethod('AddExceptionCode',Scanplus_AddExceptioncode,'Adds an exception code to the current parcel');
```
When I call method "IsParcelBlocked" from within a Python script, I get the error message "python object does not support vectorcall".

As soon as I comment one of the other methods (so that I register only 8 of them), everything works fine.

The error does also not occur when I change the following constants in PythonEngine.pas from 10 to 100:

  PYT_METHOD_BUFFER_INCREASE = 100;
  PYT_MEMBER_BUFFER_INCREASE = 100;
  PYT_GETSET_BUFFER_INCREASE = 100;

So I guess that something goes wrong in method TMethodsContainer.ReallocMethods (also in PythonEngine.pas), because with the new value of 100, ReallocMethods is never called:

```
procedure TMethodsContainer.ReallocMethods;
begin
  Inc( FAllocatedMethodCount, PYT_METHOD_BUFFER_INCREASE );
  ReAllocMem( FMethods, SizeOf(PyMethodDef)*(FAllocatedMethodCount+1));
  FillChar( FMethods[MethodCount+1] ,SizeOf(PyMethodDef)*PYT_METHOD_BUFFER_INCREASE,0);
end;

```
At first sight the method looks ok, my guess is: when there are pointers beside FMethods, after ReAllocMem they will still be pointing to the old memory locations and not to the new position of FMethods. Of course that's only a guess, the issue might be caused by something else.

**Comment by pyscripter** [comment]
- Are you using the latest sources from this repo?
- Which python version are you using?  Which platform? 
- Which Delphi version are you using?

Ideally, could you provide a minimal project that demonstrates the issue?

**Comment by pyscripter** [comment]
@lmbelo @Priyatham @jimmckeeth 

The Getit version which I assume is produced from the Embarcadero fork appears to have bugs.   Probably you need to sync again.

**Comment by lmbelo** [comment]
@pyscripter we have the latest commits synchronized to our repo. When was this issue fixed?

**Comment by pyscripter** [comment]
@lmbelo @jokli reports that this error occurs with the GetIt version but not with the current P4D master.   I don't even know how the Getit version gets generated.

@jokli Could you please try with the Embarcadero fork of P4D and see whether your error occurs or not?

**Comment by jokli** [comment]
That's really strange: after I added a lot of more code, I can't reproduce the issue anymore. I installed the sources via GetIt again to have a clean installation, but still the issue does not occur. THen I made a folder comparison between the "Source" folder from the repo and the "Source" folder via GetIt: it seems that all files are identical.

So for the moment the issue has evaporated by magic. I will notice you when I can reproduce it again.
